src/handle-data.js

```diff
@@ -31,10 +39,31 @@ export function getColumnInfo(qHyperCube, colIndex) {
       stylingInfo: info.qAttrExprInfo.map((expr) => expr.id),
       sortDirection: directionMap[info.qSortIndicator],
       dataColIdx: colIndex,
+      totalInfo,
     }
   );
 }
 
+/**
+ * Get the total head position of the table
+ *
+ * @param {Object} layout
+ * @returns the position as a string, it can be any of top, bottom or noTotals
+ */
+export function getTotalPosition(layout) {
+  let showTotal = true;
+  const hasOnlyMeasure = layout.qHyperCube.qDimensionInfo.length === 0;
+  const hasGrandTotal = layout.qHyperCube.qGrandTotalRow.length > 0;
+  const hasMeasure = layout.qHyperCube.qMeasureInfo.length > 0;
+  const isTotalModeAuto = layout.totals?.show;
+
+  if (hasOnlyMeasure || !(hasMeasure && hasGrandTotal)) showTotal = false;
+  if (!showTotal) return 'noTotals';
+  if (isTotalModeAuto || (!isTotalModeAuto && layout.totals.position === 'top')) return 'top';
+  if (!isTotalModeAuto && layout.totals.position === 'bottom') return 'bottom';
+  return 'noTotals';
```

Comment:
as I said, you don't need the showTotal variable at all. This would be even niced actually:
```suggestion
  const hasDimension = layout.qHyperCube.qDimensionInfo.length > 0;
  const hasGrandTotal = layout.qHyperCube.qGrandTotalRow.length > 0;
  const hasMeasure = layout.qHyperCube.qMeasureInfo.length > 0;
  const isTotalModeAuto = layout.totals?.show;

  if (hasDimension && hasMeasure && hasGrandTotal) {
    if (isTotalModeAuto || (!isTotalModeAuto && layout.totals.position === 'top')) return 'top';
    if (!isTotalModeAuto && layout.totals.position === 'bottom') return 'bottom';
  }
  return 'noTotals';
```

Reply:
Yes, it is not needed also something I've changed here that seems I was wrong about it. When the table has only measure and it is not in auto mode the totals still are shown. This is how it works on the old table. I've fixed it in my coming commit.

Reply:
ok, good find!